functions/ConvertTo-DbaTimeline.ps1

```diff
@@ -0,0 +1,191 @@
+function  ConvertTo-DbaTimeline {
+    <#
+        .SYNOPSIS
+            Converts InputObject to a html timeline using Google Chart
+
+        .DESCRIPTION
+            This function accepts input as pipeline from the following psdbatools functions:
+                Get-DbaAgentJobHistory
+                Get-DbaBackupHistory
+                (more to come...)
+            And generates Bootstrap based, HTML file with Google Chart Timeline
+
+        .PARAMETER InputObject
+
+            Pipe input, must an output from the above functions.
+
+        .NOTES
+            Tags: Internal
+            Author: Marcin Gminski (@marcingminski)
+
+            Dependency: ConvertTo-JsDate, Convert-DbaTimelineStatusColor
+            Requirements: None
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+-           License: MIT https://opensource.org/licenses/MIT
+
+        .LINK
+            https://dbatools.io/ConvertTo-DbaTimeline
+
+        .EXAMPLE
+            Get-DbaAgentJobHistory -SqlInstance sql-1 -StartDate ‘2018-08-13 00:00’ -EndDate ‘2018-08-13 23:59’ -NoJobSteps | ConvertTo-DbaTimeline | Out-File C:\temp\DbaAgentJobHistory.html -Encoding ASCII
+            Get-DbaBackupHistory -SqlInstance sql-1 -Since ‘2018-08-13 00:00’ | ConvertTo-DbaTimeline | Out-File C:\temp\DbaBackupHistory.html -Encoding ASCII
+
+    #>
+
+    [CmdletBinding()]
+    Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [object[]]$InputObject
+    )
+    begin {
+        #need to capture calling process to know what we are being asked for i.e. JobHistory, BackupHistory etc?
+        #I dont know of any way apart from Get-PSCallStack but that return the whole stack but in order to the last
+        #function should be the one that called this one? Not sure if this is correct but it works.
+        $caller = Get-PSCallStack | Select-Object -Property * | Select-Object -last 1
+        #check if we can handle the input
+        if ($caller.Position -NotLike "*Get-DbaAgentJobHistory*" -and `
```

Comment:
I'd like to support the possibility of someone assigning stuff to a variable then piping that in, or even using Out-GridView which I don't think this would support. In order to do that, I'll be checking the input type.